src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -46,6 +87,28 @@ protected override async Task<bool> IsSemanticTriggerCharacterAsync(Document doc
             return true;
         }
 
+        private bool ShouldTriggerInArgumentLists(SourceText text, OptionSet options)
+        {
+            if (Workspace.TryGetWorkspace(text.Container, out var workspace))
+            {
+                return ShouldTriggerInArgumentLists(workspace, options);
+            }
+
+            return false;
+        }
+
+        private bool ShouldTriggerInArgumentLists(Workspace workspace, OptionSet options)
+        {
+            var isTriggerInArgumentListOptionEnabled = options.GetOption(CompletionOptions.TriggerInArgumentLists, LanguageNames.CSharp);
+            if (isTriggerInArgumentListOptionEnabled != null)
+            {
+                return isTriggerInArgumentListOptionEnabled.Value;
+            }
+
+            var experimentationService = workspace.Services.GetService<IExperimentationService>();
+            return experimentationService.IsExperimentEnabled(WellKnownExperimentNames.TriggerCompletionInArgumentLists);
```

Comment:
god, i hope this is non-blocking on the UI thread.

Reply:
I believe this to be the case. I'll also cache the value (because I see other people doing that), but I'll defer to @JoeRobich for the real answer. :)

Reply:
Cached.